src/endpoints/query/impl.cpp

```diff
@@ -162,6 +162,21 @@ namespace
 
                 auto conn = irods::get_connection(client_info->username);
 
+                // Enable ticket if the request includes one.
+                if (const auto iter = args.find("ticket"); iter != std::end(args)) {
+                    if (const auto ec = irods::enable_ticket(conn, iter->second); ec < 0) {
+                        res.result(http::status::internal_server_error);
+                        res.body() = json{
+                            {"irods_response", {
+                                {"error_code", ec},
+                                {"error_message", "Error enabling ticket on connection."}
+                            }}
+                        }.dump();
+                        res.prepare_payload();
+                        return _sess_ptr->send(std::move(res));
+                    }
+                }
```

Comment:
Should the query endpoint accept tickets?

`iquest` doesn't expose a way to enable tickets, but it could. Perhaps there's a reason behind it not exposing a ticket option.

Reply:
After some discussion, we've concluded the /query endpoint should not accept tickets.

Only the following operations should accept tickets.
- Stat'ing a data object
- Stat'ing a collection
- Writing to a data object
- Reading a data object

Reply:
Done